hazelcast/src/test/java/com/hazelcast/map/impl/tx/MapTransactionTest.java

```diff
@@ -982,6 +982,46 @@ public Boolean execute(TransactionalTaskContext context) throws TransactionExcep
         assertTrue(b);
     }
 
+    // ========================= containsValue =====================
+    @Test(expected = NullPointerException.class)
+    public void testContainsValue_whenNullValue() throws TransactionException {
+        final HazelcastInstance hz = createHazelcastInstance();
+
+        hz.executeTransaction(options, new TransactionalTask<Boolean>() {
+            public Boolean execute(TransactionalTaskContext context) throws TransactionException {
+                TransactionalMap<Object, Object> txMap = context.getMap("default");
+                txMap.containsValue(null);
+                return true;
+            }
+        });
+    }
+
+    @Test
+    public void testContainsValue() throws TransactionException {
+        Config config = getConfig();
```

Comment:
You can switch the map in-memory format to OBJECT to see why we need the `recordComparator` like the one in `com.hazelcast.map.impl.recordstore.DefaultRecordStore#containsValue` : `config.getMapConfig("default").setInMemoryFormat(InMemoryFormat.OBJECT);`